test/services/TwoStepVerificationServiceSpec.scala

```diff
@@ -206,10 +206,12 @@ class TwoStepVerificationServiceSpec extends UnitSpec with MockitoSugar with Wit
         verify(ruleContext, Mockito.atMost(2)).enrolments
         verify(twoStepVerificationThrottleMock).isRegistrationMandatory(expectedRuleName, userId)
         verify(auditContext, atLeastOnce()).setRoutingReason(any[RoutingReason.RoutingReason], anyBoolean())(any[ExecutionContext])
-        if (isMandatory) {
-          verify(auditContext).setSentToMandatory2SVRegister(expectedRuleName)
-        } else {
-          verify(auditContext).setSentToOptional2SVRegister(expectedRuleName)
+        if (destinationIsUplifted) {
+          if (isMandatory) {
+            verify(auditContext).setSentToMandatory2SVRegister(expectedRuleName)
+          } else {
+            verify(auditContext).setSentToOptional2SVRegister(expectedRuleName)
+          }
```

Comment:
This test has a lot of re-use, which is good, but... and feel free to ignore this... seeing the nested if-statements inside the test looks like a code smell - one test is trying to do too much. I think it could be split so that there is:
- one scenario table for  destinationIsUplifted == true && isMandatory == true, 
- one scenario table for destinationIsUplifted = true && isMandatory == false, 
- and scenario table for one for destinationIsUplifted == false